backend/manager/modules/restapi/types/src/main/java/org/ovirt/engine/api/restapi/types/MigrationOptionsMapper.java

```diff
@@ -42,6 +65,40 @@ public static void copyMigrationOptions(MigrationOptions model, HasMigrationOpti
             entity.setMigrateEncrypted(mapFromInheritableBoolean(model.getEncrypted()));
         }
 
+        if (model.isSetParallelMigrationsPolicy()) {
+            Integer parallelMigrations = null;
+            final Integer customParallelMigrations = model.getCustomParallelMigrations();
+            if (model.getParallelMigrationsPolicy() != ParallelMigrationsPolicy.CUSTOM
+                    && customParallelMigrations != null) {
+                throw new IllegalArgumentException("The number of connections specified although not set to custom");
+            }
+            switch (model.getParallelMigrationsPolicy()) {
+            case INHERIT:
+                break;
+            case AUTO:
+                parallelMigrations = -2;
+                break;
+            case AUTO_PARALLEL:
+                parallelMigrations = -1;
+                break;
+            case DISABLED:
+                parallelMigrations = 0;
+                break;
+            case CUSTOM:
+                if (customParallelMigrations == null) {
+                    throw new IllegalArgumentException("The number of custom parallel migrations not specified");
+                }
+                if (customParallelMigrations < 2 || customParallelMigrations > 255) {
+                    throw new InvalidValueException("The number of custom parallel migrations is out of range (2-255)");
+                }
```

Comment:
Done.

Reply:
I think that there should be no exception thrown from the mapper - just set the values and let the command validate it. If there is a custom value set but the policy is not custom, I'd ignored that custom value.

Reply:
You're right. We could set it to e.g. Integer.MIN_VALUE to mark it as invalid and let it fail in the command. 

I am not an expert on REST, it just seems inconsistent with the rest of mappers. @liranr23, WDYT?

Reply:
> We could set it to e.g. Integer.MIN_VALUE to mark it as invalid and let it fail in the command. 

That's a good idea, this would be transparent to the user and provide a meaningful error message in the response.

Reply:
We do throw sometimes exceptions in the mappers (for example: `VmBaseMapper::mapResumeBehavior()`).
The question is the placement of the failure. Here it will fail on the REST-API call, before reaching backend. Otherwise, backend should fail and return the result, which will be shown everywhere (API response, UI if it possible, engine log).

The better approach in my opinion is backend. But, the inputs here are problematic to select the right policy which make the current API validation as a solution.

Reply:
> > If there is a custom value set but the policy is not custom, I'd ignored that custom value.
> 
> That's what we already do here.

Actually not.  But now yes :-).

> The better approach in my opinion is backend.

OK, done as Lucia suggested.

Reply:
> OK, done as Lucia suggested.

OK, so now you also need to consider the `Add*` commands and validate there also as Lucia wrote :)

Reply:
this thread has diverged - the original issue was resolved